volatility/framework/plugins/windows/memdump.py

```diff
@@ -0,0 +1,76 @@
+# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+from typing import List
+import sys
+from volatility.framework import exceptions, renderers, interfaces
+from volatility.framework.configuration import requirements
+from volatility.framework.renderers import format_hints
+from volatility.framework.objects import utility
+from volatility.plugins.windows import pslist, dlllist, ssdt, vadinfo
+
+
+class Memdump(interfaces.plugins.PluginInterface):
+    """Dump the addressable memory for a process"""
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        # Since we're calling the plugin, make sure we have the plugin's requirements
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]),
+            requirements.SymbolTableRequirement(name = "nt_symbols", description = "Windows kernel symbols"),
+            requirements.PluginRequirement(name = 'pslist', plugin = pslist.PsList, version = (1, 0, 0)),
+            requirements.PluginRequirement(name = 'vadinfo', plugin = vadinfo.VadInfo, version = (1, 0, 0)),
+            requirements.IntRequirement(name = 'pid',
+                                        description = "Process ID to include (all other processes are excluded)",
+                                        optional = True)
+        ]
+
+    def _generator(self, procs):
+        for proc in procs:
+            data = b""
+            process_name = proc.ImageFileName.cast("string",
+                                                   max_length = proc.ImageFileName.vol.count,
+                                                   errors = 'replace')
+            pid = "Unknown"
+            try:
+                pid = proc.UniqueProcessId
+                proc_layer_name = proc.add_process_layer()
+                proc_layer = self.context.layers[proc_layer_name]
+            except exceptions.InvalidAddressException as excp:
+                vollog.debug("Process {}: invalid address {} in layer {}".format(pid, excp.invalid_address, excp.layer_name))
+                continue
+
+            #Create file for writing
+            filedata = interfaces.plugins.FileInterface("{}.dmp".format(proc.UniqueProcessId))
```

Comment:
To avoid issues with pid re-use (this isn't common, but has been known to happen)...we should use another factor in addition to pid as the file names. For example, we could add `proc.vol.offset` which will be unique throughout the memory sample. In that case, if a process terminates but stays in the process list, and the another process starts with the same pid as the old one, we won't need to worry about file name conflicts.